exercises/practice/perfect-numbers/.approaches/if-statements/content.md

```diff
@@ -41,39 +40,42 @@ if (number < 1)
 ```
 
 The next step is to calculate the sum of the number's factors.
-A factor is a number that you can divide another number with and not end up with a remainder, which we can check using the modulo (`%%`) operator: `number % factor == 0`.
+A factor is a number that you can divide another number with and not end up with a remainder, which we can check using the modulo (`%`) operator: `number % factor == 0`.
 
 We then need to decide which numbers could possibly be factors.
 For the lower bound, we can use `1`, as that is always the smallest factor.
 For the upper bound, we can leverage the fact that the second lowest factor is `2`, which means that any numbers greater than `number / 2` cannot be factors.
 
-We can use the [`Enumerable.Range() method][enumerable-range] to iterate over the possible factors, then use [`Where()`][enumerable-where] to select only valid factors and then finally sum them via the [`Sum() method`][enumerable-sum]:
+We can use the [Enumerable.Sum() method][enumerable-sum] to sum all the factors, but since not all numbers are factors we need to deal with the numbers that aren't.
+Providing the [Enumerable.Sum() method][enumerable-sum] with a function tells it to keep every factor and use a `0` in place of every number that isn't, effectively disregarding it when summing.
 
 ```csharp
-var sumOfFactors = Enumerable.Range(1, number / 2)
-    .Where(factor => number % factor == 0)
-    .Sum();
+var sum = Enumerable.Range(1, number / 2)
+    .Sum(n => number % n == 0 ? n : 0);
 ```
 
 Then finally we can apply the logic to classify the perfect number via some `if` statements:
 
 ```csharp
-if (sumOfFactors < number)
+if (sum < number)
     return Classification.Deficient;
 
-if (sumOfFactors > number)
+if (sum > number)
     return Classification.Abundant;
 
 return Classification.Perfect;
 ```
 
+## Sum vs. Where and Sum
+The line `.Sum(n => number % n == 0 ? n : 0);` can also be written as `.Where(n => number % n == 0).Sum();`, using [Enumerable.Where() method][enumerable-where] to _keep_ only the factors and then [Enumerable.Sum() method][enumerable-sum] the result, but this is a little more verbose.
```

Comment:
I feel like this is the wrong way around. In other approaches, I tend to start with the code that I then refactor. That way you show to the student how things can be improved.

Reply:
Good point, I'll look into this. I'll also fix this in mentioned PR for other approach so it's more cohesive.